cli/pack/common.go

```diff
@@ -257,6 +269,23 @@ func prepareBundle(cmdCtx *cmdcontext.CmdCtx, packCtx *PackCtx,
 	if err != nil {
 		return "", err
 	}
+
+	var appNameList []string
+	for _, app := range appList {
+		appNameList = append(appNameList, app.Name)
+	}
+
+	if packCtx.IntegrityPrivateKey != "" {
+		if !packCtx.CartridgeCompat {
+			err = signer.Sign(basePath, appNameList)
+			if err != nil {
+				return "", err
+			}
+		} else {
+			return "", errors.New("cannot pack with integrity checks in cartridge-compat mode")
+		}
+	}
```

Comment:
And use it without any condition.
```suggestion
	if packCtx.IntegrityPrivateKey != "" && packCtx.CartridgeCompat {
		return "", errors.New("cannot pack with integrity checks in cartridge-compat mode")
	}

	err = signer.Sign(basePath, appNameList)
	if err != nil {
		return "", err
	}
```
Up to you. But from the code reader side it's better to choose.

Reply:
I don't like the idea of creating and making calls to `signer` regardless of whether the key is specified because, even though it works nicely with dummy implementation, it would require us to handle cases where key is not specified on the `tt-ee` part, where there is a non-trivial signer. But if you are comfortable with this, I add the suggested change.

Reply:
As an alternative: you could remove the dummy implementation and return an error from NewSingner() bacause it should not be called in the CE part.

Reply:
> you could remove the dummy implementation and return an error from NewSingner() 

Yep, that seems to me like a better thing to do. Already in progress

Reply:
We will have to ditch the `Sign` tests then, as we won't be able to obtain a signer to call it with

Reply:
Now the error is thrown. The good thing is we can now ditch `dummySigner` altogether!